prescient_sdk/client.py

```diff
@@ -0,0 +1,204 @@
+import logging
+import datetime
+import urllib.parse
+
+import msal
+import boto3
+
+from prescient_sdk.config import Settings
+
+logger = logging.getLogger("prescient_sdk")
+
+
+class PrescientClient:
+    """
+    Client for interacting with the Prescient API.
+
+    This client is used to authenticate using Azure AD, and for obtaining AWS
+    credentials using the Azure AD access token. The client also provides helpers such as
+    provding the STAC URL for the Prescient API, and authentication headers for making
+    requests to the STAC API.
+
+    Token expiration (both AWS and Azure) is handled automatically for long running instances
+    of the client (in a notebook for example).
+
+    Default configuration is set using the default values from prescient_sdk.config.
+    These can be overridden by setting values explicitly when initializing the client.
+
+    Args:
+        settings (Settings, optional): Configuration settings for the client. Defaults to None.
+
+    """
+
+    def __init__(
+        self,
+        settings: Settings | None = None,
+    ):
+        # default configuration values are set in the Settings class (prescient_sdk.config)
+        if settings is None:
+            settings = Settings()
+        self.settings: Settings = settings
+
+        # initialize empty credentials
+        self._azure_credentials: dict = {}
+        self._aws_credentials: dict = {}
+
+    @property
+    def catalog_url(self) -> str:
+        """
+        Get the STAC URL.
+
+        Returns:
+            str: The STAC URL.
+        """
+        return urllib.parse.urljoin(self.settings.endpoint_url, "stac")
+
+    @property
+    def azure_credentials(self) -> dict:
```

Comment:
Future ticket: we want our client to be able to switch between auth providers depending on customer. So this should be a generic `get_token` or `get_crededtials` function. Future logic for switching between auth provider SDK is TBD.